lib/resty/redis.lua

```diff
@@ -107,7 +65,7 @@ end
 
 
 function _M.set_timeout(self, timeout)
-    local sock = self.sock
+    local sock = rawget(self, "sock")
```

Comment:
Anyway, we need to make `rawget` a local variable on the toplevel of this module. Otherwise it is a global variable lookup.